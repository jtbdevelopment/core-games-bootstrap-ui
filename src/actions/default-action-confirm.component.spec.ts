import {TestBed} from '@angular/core/testing';
import {NgbActiveModal} from '@ng-bootstrap/ng-bootstrap';
import {DefaultActionConfirmComponent} from './default-action-confirm.component';


class MockModal {
    close = jasmine.createSpy('close');
    dismiss = jasmine.createSpy('dismiss');
}

describe('Component:  default action confirm component', () => {
    let fixture;
    let modal;
    beforeEach(() => {
        TestBed.configureTestingModule({
            declarations: [
                DefaultActionConfirmComponent,
            ],
            providers: [
                {provide: NgbActiveModal, useClass: MockModal}
            ],
        });
        TestBed.compileComponents();
        fixture = TestBed.createComponent(DefaultActionConfirmComponent);
        modal = TestBed.get(NgbActiveModal);
        fixture.detectChanges();
    });

    it('displays confirm message', () => {
        fixture.componentInstance.confirmMessage = 'This will be very bad!';
        fixture.detectChanges();
        expect(fixture.nativeElement.querySelector('.modal-title').textContent.trim()).toEqual('This will be very bad!');
    });

    it('cancel modal', () => {
        expect(modal.dismiss).not.toHaveBeenCalled();
        fixture.nativeElement.querySelector('.cancel-button').click();
        expect(modal.dismiss).toHaveBeenCalled();
    });

    it('confirm modal', () => {
        expect(modal.close).not.toHaveBeenCalled();
        fixture.nativeElement.querySelector('.action-button').click();
        expect(modal.close).toHaveBeenCalled();
    });
});
