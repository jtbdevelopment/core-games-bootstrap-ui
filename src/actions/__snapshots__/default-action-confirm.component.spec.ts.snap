// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:  default action confirm component displays confirm message 1`] = `
<ngbd-modal-content
  activeModal={[Function MockModal]}
  confirmMessage={[Function String]}
>
  <div
    class="game-confirm-dialog"
  >
    
    
    <div
      class="modal-header"
    >
      
      
      <h4
        class="modal-title"
      >
        This will be very bad!
      </h4>
      
    
    </div>
    
    
    <div
      class="modal-body"
    >
      
      
      <span
        class="confirm-message"
      >
        Are you sure?
      </span>
      
    
    </div>
    
    
    <div
      class="modal-footer"
    >
      
      
      <button
        class="btn btn-danger action-button"
      >
        Yes
      
      </button>
      
      
      <button
        class="btn btn-default focus cancel-button"
      >
        No
      
      </button>
      
    
    </div>
    
  
  </div>
  
</ngbd-modal-content>
`;
